### ISSUE TITLE
Unknown redirect with Docker

### ISSUE THREAD

**Issue by kmahyyg** [sourcegraph/sourcegraph#807]
- **Sourcegraph version:** 2.13.0
- **Platform information:** 

Docker 17.05.0
Running on Synology DSM 6.1

#### Steps to reproduce:

1. Create a docker container by using your image and DSM's docker
2. Go to `http://<DSM IP>:<Mapped port>`

#### Expected behavior:

Works fine.

#### Actual behavior:

Jump to `http://127.0.0.1:7080`

PS: Due to limitation of DSM, `/var/run/docker.sock` cannot be mapped into the inside of docker.

**Comment by lalau** [comment]
Same here. I follow the aws deploy step option A here and got the same redirct:
https://github.com/sourcegraph/sourcegraph/blob/master/doc/admin/install/docker/aws.md

**Comment by sqs** [comment]
(The /var/run/docker.sock mapping will no longer be needed in Sourcegraph 3.0.)

**Comment by nicksnyder** [comment]
@kmahyyg @lalau 

Can you share your site configuration with any secrets redacted?

I am specifically interested in what you have configured for these values:
- [appURL](https://docs.sourcegraph.com/admin/site_config/all#appurl-string)
- [canonicalURLRedirect](https://docs.sourcegraph.com/admin/site_config/all#canonicalurlredirect-string-enum)

If you haven't configured canonicalURLRedirect, can you try setting it to "disabled" to see if that fixes the issue?

We changed the default value recently https://github.com/sourcegraph/sourcegraph/pull/467

**Comment by kmahyyg** [comment]
I do not modify any config, I just run the container using the commands on your document page.

**Comment by nicksnyder** [comment]
@kmahyyg Can you try setting canonicalURLRedirect to "disabled" in your site config and let me know if that fixes the problem?

**Comment by kmahyyg** [comment]
> @kmahyyg Can you try setting canonicalURLRedirect to "disabled" in your site config and let me know if that fixes the problem?

Works fine after add ` "experimentalFeatures": {"canonicalURLRedirect":"disabled"}` .

But there's still some problem here: When generate a new password reset link: It still will jump to 7080 port instead of mapped port.

**Comment by nicksnyder** [comment]
Thanks for confirming, we will work on a fix.

**Comment by VistaarJ** [comment]
Also having the same problem. Was running sourcegraph using the quickstart installation "docker run   --publish 7080:7080 --rm   --volume ~/.sourcegraph/config:/etc/sourcegraph   --volume ~/.sourcegraph/data:/var/opt/sourcegraph   --volume /var/run/docker.sock:/var/run/docker.sock   sourcegraph/server:2.13.3" , but accessing the URL is being redirected to 127.0.0.1. I tried adding "experimentalFeatures": {"canonicalURLRedirect":"disabled"} in ~/.sourcegraph/config but still same problem.

**Comment by sqs** [comment]
@kmahyyg @VistaarJ Please try setting the `appURL` site config property to the URL you want to use, such as:

```
    "appURL": "http://1.2.3.4:1234"
```

or

```
    "appURL": "http://sourcegraph.example.com"
```

Does that fix the problem for you?

(Note: In the next (not-yet-released) version and in the latest docs, this has been renamed to `externalURL`. But for your version, it is still called `appURL`.)

**Comment by kmahyyg** [comment]
I've already migrated to the latest tag of docker image, with the original config I modified as above. Sorry, I'm a little busy these days, no time to do these test.

Sent with ProtonMail Secure Email.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by cvanlabe** [comment]
I have tried to add "appURL": "myfqdn" in ~/.sourcegraph/config/sourcegraph-config.json, and now it shows on the container output it's ready at my fqdn, however when using the browser to it, it still redirects to 127.0.0.1:7080 .

Thanks! :)

**Comment by sqs** [comment]
We just released `sourcegraph/server:2.13.4` to fix this issue ([changelog](https://about.sourcegraph.com/changelog#2134)). Use that Docker image tag (instead of `sourcegraph/server:2.13.3`). It's up at https://docs.sourcegraph.com/ now, too. Sorry all for the inconvenience!

@cvanlabe I believe that redirect may be caused by your browser remembering the permanent redirect sent by Sourcegraph. Try clearing your browser cache for your preferred hostname. If you can't even navigate to it to clear the cache without being redirected, go to `http://externalurl:7080/__version` (which should not trigger a redirect) and then clear the cache. If that doesn't work, try it in incognito mode and let me know if that works. Sorry - it should work once the cache is cleared, though!

**Comment by cvanlabe** [comment]
Perfect, it works flawlessly with 2.13.4. 
2.13.3 worked with the workaround indeed (you were right, browser was acting smartypants).

One last suggestion now: update the install doc to not explicitly point to localhost but to the < IP of the machine >, but that's completely up to you.

Thanks for the quick turnaround!
